arch/x86/slide_avx.c

```diff
@@ -18,7 +18,7 @@ ZLIB_INTERNAL void slide_hash_avx2(deflate_state *s) {
     Pos *p;
     unsigned n;
     unsigned wsize = s->w_size;
-    const __m256i zmm_wsize = _mm256_set1_epi16(wsize);
+    const __m256i zmm_wsize = _mm256_set1_epi16((uint16_t)wsize);
```

Comment:
Because wsize is local variable, cast here seems wrong solution.

Reply:
Why?

Reply:
Elsewhere `wsize` is assigned to `n` which is `unsigned` type..

Reply:
Back when I started adding support for PowerPC, I suggested changing the type of ```s->w_size``` to ```uint16_t``` as all optimized versions want it to be 16 bits, not 32 bits... It wasn't changed back then because it caused build errors on some platforms due to overflow somewhere... I still think the variable size should be changed as early as possible...

Reply:
If ```s->hash_size``` can't be more than what fits in ```uint16_t``` then ```n``` can also be ```uint16_t```.

Reply:
It seems it can be bigger..

Reply:
Or maybe not since MAX_MEM_LEVEL is 8. I think it might be better to change all that when w_size is changed to uint16.

Reply:
> I still think the variable size should be changed as early as possible...

I have added another commit that should do just that.